core/sdk/src/test/java/com/mongodb/stitch/core/auth/internal/CoreStitchAuthUnitTests.java

```diff
@@ -577,6 +580,58 @@ public void testProfileRequestFailureEdgeCases() {
     assertEquals(userToBeLinked.getId(), auth.getUser().getId());
   }
 
+  @Test
+  public void testSwitchUser() {
+    final StitchRequestClient requestClient = getMockedRequestClient();
+    final StitchAuthRoutes routes = new StitchAppRoutes("my_app-12345").getAuthRoutes();
+    final StitchAuth auth = new StitchAuth(
+        requestClient,
+        routes,
+        new MemoryStorage());
+
+    try {
+      auth.switchUser("not_a_user_id");
+      fail("should have thrown error due to missing key");
+    } catch (IllegalArgumentException e) {
+      assertNotNull(e);
+    }
+
+    final CoreStitchUser user =
+        auth.loginWithCredentialInternal(new AnonymousCredential());
+
+    assertEquals(user, auth.switchUser(user.getId()));
+    assertEquals(user, auth.getUser());
+
+    final CoreStitchUser user2 =
+        auth.loginWithCredentialInternal(new UserPasswordCredential("hello ", "friend"));
+
+    assertEquals(user2, auth.getUser());
+    assertNotEquals(user, auth.getUser());
+
+    assertEquals(user, auth.switchUser(user.getId()));
+    assertEquals(user, auth.getUser());
```

Comment:
[task] test to see that the logout case is handled correctly as well. Logout with two logged-in users should set active user to the other user, and logout with one logged-in user should result in no active user, and isLoggedIn to be false.